approval_polls/views.py

```diff
@@ -1,101 +1,212 @@
+class PollUpdate(generic.View):
+    '''A generic update view for our poll.
+
+    Since this is actually a compound form updating two different models
+    a Poll and "n" number of Choice instances we cannot use the straight forward
+    UpdateView.
+    '''
+
+
+    success_url = '/approval_polls/my-polls'
+
+    @method_decorator(login_required)
+    def get(self, request, pk):
+        '''We have to get the object and the choices.
+
+        Arguments:
+            request (WSGIRequest): the standard request that django calls the get method with
+            pk (int): the id of the poll object we are mean to edit
+
+        Returns:
+            HTTPResponse
+        '''
+
+        form = PollUpdateForm(instance=Poll.objects.get(pk=pk))
+
+        choices = [{'form': ChoiceForm(instance=choice), 'id': choice.id} for choice in Choice.objects.filter(poll__pk=pk)]
+        my_context = {'form': form, 'choices': choices}
+        request_context = RequestContext(request)
+
+        return render_to_response('approval_polls/poll_update_form.html',
+                                  my_context, context_instance=request_context)
+
+    @method_decorator(login_required)
+    def post(self, request, pk):
+        '''Recieving the updated data for the Poll and it's choices
+
+        Arguments:
+            request (WSGIRequest): The request.POST member will contain all the data
+            from the form. This must be separated out and validated manually.
+
+            pk (int): The primary key of the Poll being updated.
+
+        Returns:
+            WSGIResponse: Upon success it redirects to "my-polls". Upon any failure
+            it returns to the poll editing page with the information pre-filled and any
+            error messages.
+        '''
+
+        # First get the data from the post and sort it out.
+        for k in ['question', 'open_date', 'close_date', 'suspended']:
+            try:
+                poll_data[k] = request.POST.get(k)
+            except NameError:
+                poll_data = {k: request.POST.get(k)}
+
+        for k in request.POST.keys():
+            if k.startswith('choice_text'):
+                # We have a choice
+                choice_id = k.split('.')[1]
+                choice_text = request.POST.get(k)
+                try:
+                    choices.append({'choice_text': choice_text, 'id': choice_id})
+                except NameError:
+                    choices = [{'choice_text': choice_text, 'id': choice_id}]
+
+        # Pass the data for the Poll into PollForm and validate it.
+        poll_form = PollUpdateForm(data=poll_data, instance=Poll.objects.get(pk=pk))
+        poll_valid = poll_form.is_valid()
+
+        # For each Choice pass the data into ChoiceForm and validate.
+
+        try:
+            choice_forms = []  # this is ugly but I will do it for now.
+            for choice in choices:
+                choice_form = ChoiceForm(data={'choice_text': choice['choice_text']},
```

Comment:
Should we also validate the choice_form? Choices may be blank and blank choices should not be allowed.